daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -184,6 +185,38 @@ func (l *logStream) Close() error {
 
 // create creates a log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+
+	// Create log group first
+	{
```

Comment:
I think you can take it out of the block with a few minor tweaks.  Specifically:
* Some of the `:=`s in the original code (not what you added) will probably need to change to `=` as Go makes a distinction between declaration and assignment.
* You'll probably want to change both instances of `input` to have different names that are more specific to what they're doing, since a single variable cannot be multiple different types.

As I said in my first comment, I'm not too opposed to this behavior change if it solves a real problem (and it sounds like it does).  One way to change it perhaps is to enable the behavior only when a log option enables it (something like `awslogs-enable-creation=true`).